main.cpp

```diff
@@ -78,15 +74,18 @@ struct Comparator                            //4
     Comparator() {}
     T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a && b != nullptr)
```

Comment:
This is incorrect.  What you want is:
```
if( a != nullptr &&
    b != nullptr )
```
You need to be explicit with what you're comparing when you write `if()` expressions.
What you wrote is actually interpreted as `if a is anything other than 0 and b is not a nullptr` by the compiler
- [ ] revise